module/Motion/ColorStraight.cpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * @file   ColorStraight.cpp
+ * @brief  目標色まで直進するクラス
+ * @author bizyutyu
+ */
+
+#include "ColorStraight.h"
+using namespace std;
+
+ColorStraight::ColorStraight(COLOR _targetColor, double _targetSpeed)
+  : Straight(_targetSpeed), targetColor(_targetColor)
+{
+}
+
+bool ColorStraight::isMetPreCondition()
+{
+  // 目標速度値が0の場合は終了する
+  if(targetSpeed == 0) {
+    logger.logWarning("The targetSpeed value passed to ColorStraight is 0");
+    return false;
+  }
+  // 目標の色がNoneのときwarningを出して終了する
+  if(targetColor == COLOR::NONE) {
+    logger.logWarning("The targetColor passed to ColorStraight is NONE");
+    return false;
+  }
+
+  return true;
+}
+
+bool ColorStraight::isMetContinuationCondition()
+{
+  COLOR color = ColorJudge::getColor(measurer.getRawColor());
+  if(color != targetColor) {
+    return true;
+  }
+
+  return false;
+}
```

Comment:
継続条件を満たしているかの関数なので，通常の返り値をtrueにした方がよさそうです！
```suggestion
bool ColorStraight::isMetContinuationCondition()
{
  COLOR color = ColorJudge::getColor(measurer.getRawColor());
  if(color == targetColor) {
    return false;
  }

  return true;
}
```